Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import {
+  HistoryModel,
+  OrderModel,
+  PaymentModel,
+  UpdateOrderStatusModel,
+} from "../../types";
+import { authenticatedFetch, parseBadRequest } from "../../util/api";
+import { ReactTable } from "../../Shared/ReactTable";
+import { createColumnHelper } from "@tanstack/react-table";
+import OrderForm from "./OrderForm";
+import { usePermissions } from "../../Shared/usePermissions";
+import { useConfirmationDialog } from "../../Shared/ConfirmationDialog";
+import { usePromiseNotification } from "../../util/Notifications";
+import { notEmptyOrFalsey } from "../../util/ValueChecks";
+import { ShowFor } from "../../Shared/ShowFor";
+import AppContext from "../../Shared/AppContext";
+import {
+  convertToPacificDate,
+  convertToPacificTime,
+} from "../../util/DateHelper";
+
+export const Details = () => {
+  const { cluster, orderId } = useParams();
+  const [{ user }] = useContext(AppContext);
+  const [order, setOrder] = useState<OrderModel | null>(null);
+  const [balanceRemaining, setBalanceRemaining] = useState<number>(0);
+  const [balancePending, setBalancePending] = useState<number>(0);
+  const { isClusterAdminForCluster } = usePermissions();
+  const [isClusterAdmin, setIsClusterAdmin] = useState(null);
+  const [notification, setNotification] = usePromiseNotification();
+  const [updateStatusModel, setUpdateStatusModel] =
+    useState<UpdateOrderStatusModel | null>(null);
+  const adminEditableStatuses = ["Processing", "Active"];
+  const sponsorEditableStatuses = ["Created"];
+
+  useEffect(() => {
+    setIsClusterAdmin(isClusterAdminForCluster());
+  }, [isClusterAdmin, isClusterAdminForCluster]);
+
+  const calculateBalanceRemaining = (data: any) => {
+    const balanceRemaining = parseFloat(data.balanceRemaining);
+    setBalanceRemaining(balanceRemaining);
+    const balancePending = data?.payments
+      .filter(
+        (payment) =>
+          payment.status !== "Completed" && payment.status !== "Cancelled",
+      )
+      .reduce((acc, payment) => acc + parseFloat(payment.amount), 0);
+    setBalancePending(balancePending);
+  };
+
+  useEffect(() => {
+    const fetchOrder = async () => {
+      const response = await authenticatedFetch(
+        `/api/${cluster}/order/get/${orderId}`,
+      );
+
+      if (response.ok) {
+        const data = await response.json();
+        console.log(data);
+        setOrder(data);
+        calculateBalanceRemaining(data);
+      } else {
+        alert("Error fetching order");
```

Comment:
use toast/notifications?  Also in general, why sometimes you use the promise notification callback thing and sometimes not?

Reply:
#392